test/e2e/queue.go

```diff
@@ -140,6 +140,26 @@ var _ = Describe("AppWrapper E2E Test", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 
+	It("MCAD CPU Requeuing Test", func() {
+		fmt.Fprintf(os.Stdout, "[e2e] MCAD CPU Requeuing Test - Started.\n")
+
+		context := initTestContext()
+		var appwrappers []*arbv1.AppWrapper
+		appwrappersPtr := &appwrappers
+		defer cleanupTestObjectsPtr(context, appwrappersPtr)
+
+		// Create a job with init containers that need 200 seconds to be ready before the container starts.
+		// The requeuing mechanism is set to start at 1 minute, which is not enough time for the PODs to be completed.
+		// The job should be requeued 3 times before it finishes since the wait time is doubled each time the job is requeued (i.e., initially it waits
+		// for 1 minutes before requeuing, then 2 minutes, and then 4 minutes). Since the init containers take 3 minutes
+		// and 20 seconds to finish, a 4 minute wait should be long enough to finish the job successfully
+		aw := createJobAWWithInitContainer(context, "aw-job-3-init-container")
+		appwrappers = append(appwrappers, aw)
+
+		err := waitAWPodsCompleted(context, aw) // This test waits for 10 minutes to make sure all PODs complete
```

Comment:
did we check Appwrapper status? do we want appwrapper to be complete too?

Reply:
Mmm I don't know. I don't think that is a big deal as the point here is that the PODs get completed. Also I don't see any test checking for the AppWrapper to be `Complete`